pkgs/applications/misc/keepassx/community.nix

```diff
@@ -57,6 +58,11 @@ stdenv.mkDerivation rec {
   NIX_LDFLAGS = optionalString stdenv.isDarwin "-rpath ${libargon2}/lib";
 
   patches = [
+    (fetchpatch {
+      # autostart: Linux: Exec= filename not absolute path
+      url = "https://github.com/keepassxreboot/keepassxc/pull/8191.patch";
```

Comment:
```suggestion
    # autostart: Linux: Exec= filename not absolute path
    (fetchpatch {
      url = "https://github.com/keepassxreboot/keepassxc/commit/4e0a2cd39ddbdb30b9308494381e6fb494ca8481.patch";
```

Reply:
`pull/8191` makes it clear that this has not yet been committed.
`commit/4e0a2cd...` only works because GitHub makes commits available across repositories, even when my `4e0a2cd` has not yet been committed to `keepassxreboot/keepassxc` proper.

I've also put the comment inside the `fetchpatch` block because it
a) is immediately concerned about/attached to the following `url = ...` line (that's the title of the PR, hence using `pull/8191` and not a commit),
b) provides slightly better ergonomics for future hackers removing this patch on the next update (`da(` in vim conveniently zaps the entire block regardless of where the cursor is inside the parentheses).

If you strongly prefer a commit hash to a PR number, I'll wait for upstream to merge to be able to use their genuine commit hash.

Reply:
> `pull/8191` makes it clear that this has not yet been committed.

and will break if you push to the branch or rebase it.



> I've also put the comment inside the `fetchpatch` block because it

it is convention that especial for CVE fixes the comment is put before the patch which might be vendored or fetched.



> If you strongly prefer a commit hash to a PR number, I'll wait for upstream to merge to be able to use their genuine commit hash.

It must be some hash. Using only the PR number is not allowed. If it is merged or not does not really matter. Even https://github.com/keepassxreboot/keepassxc/pull/8191/commits/4e0a2cd39ddbdb30b9308494381e6fb494ca8481.patch is allowed which redirects to what I suggested.

Reply:
> > `pull/8191` makes it clear that this has not yet been committed.
> 
> and will break if you push to the branch or rebase it.

Only if the actual diff changes.
`fetchpatch` normalises before checksumming, so merely amending the commit message or rebasing where the changed file did not change will still result in the same output.

> it is convention that especial for CVE fixes the comment is put before the patch which might be vendored or fetched.
> 
> > If you strongly prefer a commit hash to a PR number, I'll wait for upstream to merge to be able to use their genuine commit hash.
> 
> It must be some hash. Using only the PR number is not allowed. If it is merged or not does not really matter. Even https://github.com/keepassxreboot/keepassxc/pull/8191/commits/4e0a2cd39ddbdb30b9308494381e6fb494ca8481.patch is allowed which redirects to what I suggested.

`git grep -E '/pull/[0-9]+\.(diff|patch)'` made the impression on me that this is in fact allowed in nixpkgs.

I like the `pull/.../commit/...` link. thanks.

Reply:
> `git grep -E '/pull/[0-9]+\.(diff|patch)'` made the impression on me that this is in fact allowed in nixpkgs.

grepping through source code does not always reflect current policies as many old things are not updated within a reasonable timeframe.

Reply:
> I like the pull/.../commit/... link. thanks.

Yes, that's a good solution.